internal/cmd/cmd.go

```diff
@@ -236,95 +257,81 @@ func getRootCommand(exitCodeAddr *int, args []string, stdin io.Reader, stdout io
-				return runner.All(args, flags.disableFormat, flags.disableLint)
-			})
+			checkCmd(exitCodeAddr, stdin, stdout, stderr, flags, func(runner exec.Runner) error { return runner.ServiceDescriptorProto(args) })
 		},
 	}
-	flags.bindDisableFormat(allCmd.PersistentFlags())
-	flags.bindDisableLint(allCmd.PersistentFlags())
-	flags.bindDirMode(allCmd.PersistentFlags())
+	flags.bindDirMode(serviceDescriptorProtoCmd.PersistentFlags())
 
-	grpcCmd := &cobra.Command{
-		Use:   "grpc dirOrProtoFiles... serverAddress package.service/Method requestData",
-		Short: "Call a gRPC endpoint.",
+	versionCmd := &cobra.Command{
+		Use:   "version",
+		Short: "Print the version.",
+		Args:  cobra.NoArgs,
 		Run: func(cmd *cobra.Command, args []string) {
-			checkCmd(exitCodeAddr, stdin, stdout, stderr, flags, func(runner exec.Runner) error {
-				return runner.GRPC(args, flags.headers, flags.callTimeout, flags.connectTimeout, flags.keepaliveTime)
-			})
+			checkCmd(exitCodeAddr, stdin, stdout, stderr, flags, exec.Runner.Version)
 		},
 	}
-	flags.bindHeaders(grpcCmd.PersistentFlags())
-	flags.bindCallTimeout(grpcCmd.PersistentFlags())
-	flags.bindConnectTimeout(grpcCmd.PersistentFlags())
-	flags.bindKeepaliveTime(grpcCmd.PersistentFlags())
-	flags.bindDirMode(grpcCmd.PersistentFlags())
 
 	rootCmd := &cobra.Command{Use: "prototool"}
-	rootCmd.AddCommand(versionCmd)
-	rootCmd.AddCommand(initCmd)
-	rootCmd.AddCommand(downloadCmd)
+	rootCmd.AddCommand(allCmd)
+	rootCmd.AddCommand(binaryToJSONCmd)
 	rootCmd.AddCommand(cleanCmd)
-	rootCmd.AddCommand(filesCmd)
 	rootCmd.AddCommand(compileCmd)
-	rootCmd.AddCommand(genCmd)
 	rootCmd.AddCommand(descriptorProtoCmd)
+	rootCmd.AddCommand(downloadCmd)
 	rootCmd.AddCommand(fieldDescriptorProtoCmd)
-	rootCmd.AddCommand(serviceDescriptorProtoCmd)
-	rootCmd.AddCommand(protocCommandsCmd)
+	rootCmd.AddCommand(filesCmd)
+	rootCmd.AddCommand(formatCmd)
+	rootCmd.AddCommand(genCmd)
+	rootCmd.AddCommand(grpcCmd)
+	rootCmd.AddCommand(initCmd)
+	rootCmd.AddCommand(jsonToBinaryCmd)
 	rootCmd.AddCommand(lintCmd)
-	rootCmd.AddCommand(listLintersCmd)
 	rootCmd.AddCommand(listAllLintersCmd)
-	rootCmd.AddCommand(listLintGroupCmd)
 	rootCmd.AddCommand(listAllLintGroupsCmd)
-	rootCmd.AddCommand(formatCmd)
-	rootCmd.AddCommand(binaryToJSONCmd)
-	rootCmd.AddCommand(jsonToBinaryCmd)
-	rootCmd.AddCommand(allCmd)
-	rootCmd.AddCommand(grpcCmd)
+	rootCmd.AddCommand(listLintersCmd)
+	rootCmd.AddCommand(listLintGroupCmd)
+	rootCmd.AddCommand(protocCommandsCmd)
+	rootCmd.AddCommand(serviceDescriptorProtoCmd)
+	rootCmd.AddCommand(versionCmd)
 
-	flags.bindDebug(rootCmd.PersistentFlags())
 	flags.bindCachePath(rootCmd.PersistentFlags())
-	flags.bindProtocURL(rootCmd.PersistentFlags())
-	flags.bindPrintFields(rootCmd.PersistentFlags())
+	flags.bindDebug(rootCmd.PersistentFlags())
 	flags.bindHarbormaster(rootCmd.PersistentFlags())
+	flags.bindPrintFields(rootCmd.PersistentFlags())
+	flags.bindProtocURL(rootCmd.PersistentFlags())
```

Comment:
It would help to document how the flags are persisted somewhere. It doesn't necessarily need to be here (and maybe it already exists somewhere), but I think this might help with potential debugging later on. As far as I understand it, the "debug" flag is bound on the `rootCmd` and is therefore passed along to all of its sub-commands (everything in this case).

Reply:
What would you propose? I can put a comment that all flags put on `rootCmd` are propagated to all sub-commands, does that work? The propagation is part of the Cobra API too.

Reply:
Yea that's fair. I guess this was mostly just a foreign library use to me and required some digging around. I think it's justifiable to leave it out for now.